### DISCUSSION TITLE
Iden position in generated migration files

### DISCUSSION THREAD

**Discussion by AngelOnFira** [SeaQL/sea-orm#1339] [Ideas]
Currently, a generated migration file looks like the following:

```rust
...

#[async_trait::async_trait]
impl MigrationTrait for Migration {
    ...
}

/// Learn more at https://docs.rs/sea-query#iden
#[derive(Iden)]
enum Post {
    Table,
    Id,
    Title,
    Text,
}
```

I'm curious as to why the `Iden` of the table is below the `impl` block. I tend to move it above, for a few reasons:

- Easier to understand in a sequential way
- Copilot only gains context from code that is before it (afaik), so having it like this helps it infer context properly
- I'm used to scrolling up to change types as needed

To be clear, this is what I'd expect:

```rust
...

/// Learn more at https://docs.rs/sea-query#iden
#[derive(Iden)]
enum Post {
    Table,
    Id,
    Title,
    Text,
}

#[async_trait::async_trait]
impl MigrationTrait for Migration {
    ...
}
```

These are definitely quite subjective, but I imagine most people are used to editing stuff in the way I'm describing. I'd be curious to hear about any reasons the `Iden` above shouldn't be the default :slightly_smiling_face:

**Comment by tyt2y3** [comment]
I don't have a strong opinion, but IMO the migration functions are the meat, where the enum definitions are just details.
I agree with "important bits first". But of course what's more important is subjective.

**Reply by AngelOnFira** [reply]
Thanks for the reply! :slightly_smiling_face: I definitely agree, the methods contain the most important items. However, they're still derived from the `Iden`, and I don't think it's about importance; I think it's about quickly gathering a mental model of the migration.

For example, if I'm reading through past migrations on a new project, I'd expect to look at the `Iden` as a "table of contents" on what to expect, especially since it (likely) won't contain fields that aren't referenced in this migration. The alternative to this would be to find the fields that are altered in the migration `impl` block, which seems more difficult.
